nexusutils/dataconverter/writer.py

```diff
@@ -202,22 +207,27 @@ def __nxdl_to_attrs(self, path: str = '/') -> dict:
 
         # Fetch values for required attributes requested by the NXDL
         for attr_name in elem.findall(f"{self.nxs_namespace}attribute"):
-            elem.attrib[attr_name.get('name')] = self.data[f"{path}/@{attr_name.get('name')}"] or ''
+            key = f"{path}/@{attr_name.get('name')}"
+            if key in self.data:
+                elem.attrib[attr_name.get('name')] = self.data[key]
+            else:
+                elem.attrib[attr_name.get('name')] = ''
 
         return elem.attrib
 
     def ensure_and_get_parent_node(self, path: str, undocumented_paths) -> h5py.Group:
         """Returns the parent if it exists for a given path else creates the parent group."""
         parent_path = path[0:path.rindex('/')] or '/'
         parent_path_hdf5 = helpers.convert_data_dict_path_to_hdf5_path(parent_path)
-        parent_undocumented_paths = [path[0:path.rindex("/")] or "/" for path in undocumented_paths]
         if not does_path_exist(parent_path, self.output_nexus):
-            parent = self.ensure_and_get_parent_node(parent_path, parent_undocumented_paths)
+            parent = self.ensure_and_get_parent_node(parent_path, undocumented_paths)
             grp = parent.create_group(parent_path_hdf5)
-            if path not in undocumented_paths:
+            try:
                 attrs = self.__nxdl_to_attrs(parent_path)
                 if attrs is not None:
                     grp.attrs['NX_class'] = attrs["type"]
+            except Exception as exc:
+                raise exc
```

Comment:
no, we do not want to raise an exception here. The only reason for calling __nxdl_to_attrs is to check if the path just created was defined in NXDL, so we can label it according, otherwise, we simply continue without labeling.  
Note that this try/except could go directly around he specific call and make attrs None if cathched.